routers/web/feed/render.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package feed
+
+import (
+	"fmt"
+
+	model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// RenderBranchFeedRss render rss format for branch feed
+func RenderBranchFeedRss(ctx *context.Context) {
+	render(ctx, "rss")
+}
+
+// RenderBranchFeedAtom render atom format for branch feed
+func RenderBranchFeedAtom(ctx *context.Context) {
+	render(ctx, "atom")
+}
+
+// RenderRepoFeed handles if an RSS feed should be rendered, injects variables into context if not.
+//
+// The logic for rendering as a rss / atom feed checks against:
+// * existence of Accept header containing application/rss+xml or application/atom+xml
+// * support for the {repo}.rss url
+func RenderRepoFeed(ctx *context.Context) {
+	if !setting.EnableFeed {
+		return
+	}
+	isFeed, _, showFeedType := GetFeedType(ctx.Params(":reponame"), ctx.Req)
+	if !isFeed {
+		ctx.Data["EnableFeed"] = true
+		ctx.Data["FeedURL"] = ctx.Repo.Repository.HTMLURL()
```

Comment:
IIRC https://github.com/go-gitea/gitea/pull/23904 already injects the `EnableFeed` and `FeedURL` data for all repo handlers

And this function is only called by `Home`, is it better to revert to old code structure?

eg:


```go
func Home(ctx *context.Context) {
	if setting.EnableFeed {
		isFeed, _, showFeedType := feed.GetFeedType(ctx.Params(":reponame"), ctx.Req)
		if isFeed {
			render(...)
			return
		}
		...
```

Reply:
Yes that makes sense. I've reverted to the old code structure.

There is still (albeit refactored) ```feed.RenderBranchFeed``` which handles rendering the feeds for branches / files. Is this ok or would you prefer to remove this function and either in-line or handle through the ```Home``` handler?

Reply:
Do you mean `render`? I guess it could be simplified a little (and in-lining seems good enough)

```go
	switch {
	case ctx.Link == fmt.Sprintf("%s.%s", ctx.Repo.RepoLink, showFeedType):
		ShowRepoFeed(....)
	case ctx.Repo.TreePath == "":
		ShowRepoFeed(....)
	default: // for files
		ShowRepoFeed(....)
	}
```

What do you think ?